mono/mini/interp/transform.c

```diff
@@ -3539,12 +3539,6 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 				goto_if_nok (error, exit);
 			}
 
-			MonoClass *tos_klass = td->sp [-1].klass;
-			if (tos_klass && td->sp [-1].type == STACK_TYPE_VT) {
-				int tos_size = mono_class_value_size (tos_klass, NULL);
-				POP_VT (td, tos_size);
-			}
-
```

Comment:
sounds like this was wrong back then? https://github.com/mono/mono/pull/5791/commits/eae0a0604a8eafe3e8793a08dd754b01418ca310

Reply:
I'd say so. I remember I put a g_assert there and it wasn't reached. Also those tests didn't seem to fail with this change. Maybe it was an accidental fix, by popping some stuff that should had been handled elsewhere